java/openmldb-spark-connector/src/main/java/com/_4paradigm/openmldb/spark/write/OpenmldbDataWriter.java

```diff
@@ -0,0 +1,154 @@
+    private PreparedStatement preparedStatement = null;
+
+    public OpenmldbDataWriter(OpenmldbWriteConfig config, int partitionId, long taskId) {
+        try {
+            SdkOption option = new SdkOption();
+            option.setZkCluster(config.zkCluster);
+            option.setZkPath(config.zkPath);
+            SqlClusterExecutor executor = new SqlClusterExecutor(option);
+            String dbName = config.dbName;
+            String tableName = config.tableName;
+
+            Schema schema = executor.getTableSchema(dbName, tableName);
+            // create insert placeholder
+            StringBuilder insert = new StringBuilder("insert into " + tableName + " values(?");
+            for (int i = 1; i < schema.getColumnList().size(); i++) {
+                insert.append(",?");
+            }
+            insert.append(");");
+            preparedStatement = executor.getInsertPreparedStmt(dbName, insert.toString());
+        } catch (SQLException | SqlException e) {
+            e.printStackTrace();
+        }
+
+        this.partitionId = partitionId;
+        this.taskId = taskId;
+    }
+
+    @Override
+    public void write(InternalRow record) throws IOException {
+        try {
+            // record to openmldb row
+            ResultSetMetaData metaData = preparedStatement.getMetaData();
+            Preconditions.checkState(record.numFields() == metaData.getColumnCount());
+            addRow(record, preparedStatement);
+            preparedStatement.addBatch();
+        } catch (SQLException e) {
+            e.printStackTrace();
+            throw new IOException(e.getMessage());
+        }
+    }
+
+    private void addRow(InternalRow record, PreparedStatement preparedStatement) throws SQLException {
+        // idx in preparedStatement starts from 1
+        ResultSetMetaData metaData = preparedStatement.getMetaData();
+        for (int i = 0; i < record.numFields(); i++) {
+            // sqlType
+            int type = metaData.getColumnType(i + 1);
+            switch (type) {
+                case Types.BOOLEAN:
+                    preparedStatement.setBoolean(i + 1, record.getBoolean(i));
+                    break;
+                case Types.SMALLINT:
+                    preparedStatement.setShort(i + 1, record.getShort(i));
+                    break;
+                case Types.INTEGER:
+                    preparedStatement.setInt(i + 1, record.getInt(i));
+                    break;
+                case Types.BIGINT:
+                    preparedStatement.setLong(i + 1, record.getLong(i));
+                    break;
+                case Types.FLOAT:
+                    preparedStatement.setFloat(i + 1, record.getFloat(i));
+                    break;
+                case Types.DOUBLE:
+                    preparedStatement.setDouble(i + 1, record.getDouble(i));
+                    break;
+                case Types.VARCHAR:
+                    preparedStatement.setString(i + 1, record.getString(i));
```

Comment:
Should we escape the single quote or double quote?

Reply:
We load file by DataFrameReader first, so DataFrameReader should do escape by add some options ,ref https://spark.apache.org/docs/3.2.0/sql-data-sources.html
I'll check how quote works in spark.

Reply:
I'm worry about if our `preparedStatement` supports passing the string with double quote. Using string to generate SQL may have this issue but normally `preparedStatement` can handle this. If so, we do need to do extra work for string with double quote.

Reply:
You mean a row like `"abc", 1, 2` and quote is `"`, we 'll get a column `"abc"`, the string includes quotes?